lib/FrameManager.js

```diff
@@ -637,6 +629,20 @@ class Frame {
       this._parentFrame._childFrames.delete(this);
     this._parentFrame = null;
   }
+
+  /**
+   * @param {function(string)} pageFunction
+   * @param {!string} arg
+   * @return {!Promise<!ElementHandle>}
+   */
+  async _evaluateHandle(pageFunction, arg) {
+    const context = await this._contextPromise;
+    this._page._securityError = null;
+    const element = await context.evaluateHandle(pageFunction, arg);
+    if (this._page._securityError)
+      throw this._page._securityError;
```

Comment:
this `securityError` is not necessarily generated by the evaluation above.
In order to link one to another, you'd need to label evaluation with unique sourceURL and match
log entries by this sourceURL.

Do you want to proceed with this? Feel free to ask me any questions.
Let me know if you don't have time to deal with this: I'd like to have this landed and I'll take this over.

Reply:
@aslushnikov
Thanks! Let me handle this for a day or two because I can learn a lot from you in this way. I'll ask you if it's difficult :)